esphome/components/wake_on_lan/button.py

```diff
@@ -0,0 +1,33 @@
+import esphome.codegen as cg
+from esphome.components import button
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+
+CONF_TARGET_MAC_ADDRESS = "target_mac_address"
+
+wake_on_lan_ns = cg.esphome_ns.namespace("wake_on_lan")
+
+WakeOnLanButton = wake_on_lan_ns.class_("WakeOnLanButton", button.Button, cg.Component)
+
+DEPENDENCIES = ["wifi"]
+
+CONFIG_SCHEMA = (
+    cv.ENTITY_BASE_SCHEMA.extend(button.BUTTON_SCHEMA)
```

Comment:
The BUTTON_SCHEMA already extends ENTITY_BASE_SCHEMA
```suggestion
    button.BUTTON_SCHEMA
```